projects/services/storage-service-typescript/test/storage-service.spec.ts

```diff
@@ -0,0 +1,85 @@
+import * as chai from "chai";
+import * as fse from "fs-extra";
+import * as path from "path";
+import * as getPort from "get-port";
+import { stubInterface } from "ts-sinon";
+
+import { types, BlockUtils, ErrorHandler, GRPCServerBuilder, grpc } from "orbs-core-library";
+import { BlockStorageClient, StateStorageClient, GossipServer } from "orbs-interfaces";
+import storageServer from "../src/server";
+import GossipService from "../../gossip-service-typescript/src/service";
+import TransactionPoolService from "../../consensus-service-typescript/src/transaction-pool-service";
+
+const { expect } = chai;
+
+ErrorHandler.setup();
+
+describe("new storage server test", function () {
+  let server: GRPCServerBuilder;
+  let blockClient: BlockStorageClient;
+  let stateClient: StateStorageClient;
+
+  before(async () => {
+    const endpoint = `127.0.0.1:${await getPort()}`;
+
+    const topology =  {
+      peers: [
+        {
+          service: "storage",
+          endpoint: endpoint,
+        },
+        {
+          service: "gossip",
+          endpoint: endpoint,
+        },
+        {
+          service: "consensus",
+          endpoint: endpoint,
+        },
+      ],
+    };
+
+    const NODE_NAME = "tester";
+    const BLOCK_STORAGE_POLL_INTERVAL = 5000;
+    const BLOCK_STORAGE_DB_PATH = "./test/db/";
```

Comment:
The code folder is not necessarily writable. The DB should be created in a temp folder.